web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/AgentWaitingDialog.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import Dialog, { DialogContent } from 'design/DialogConfirmation';
+import { AnimatedProgressBar, Box, ButtonPrimary, Text } from 'design';
+import React from 'react';
+
+import * as Icons from 'design/Icon';
+
+import { Kube } from 'teleport/services/kube';
+import { JoinToken } from 'teleport/services/joinToken';
+import { usePingTeleport } from 'teleport/Discover/Shared/PingTeleportContext';
+import { Mark, TextIcon, useShowHint } from 'teleport/Discover/Shared';
+import { HintBox } from 'teleport/Discover/Shared/HintBox';
+
+type AgentWaitingDialogProps = {
+  joinToken: JoinToken;
+  status: string;
+  clusterName: string;
+  setWaitingResult(cluster: Kube): void;
+  close(): void;
+};
+
+export function AgentWaitingDialog({
+  joinToken,
+  status,
+  clusterName,
+  setWaitingResult,
+  close,
+}: AgentWaitingDialogProps) {
+  const { result, active } = usePingTeleport<Kube>(joinToken);
+  setWaitingResult(result);
+  const showHint = useShowHint(active);
+
+  function hintMessage() {
+    if (showHint && !result) {
```

Comment:
i waited long enough for storybook to reach the default timeout and it renders this (i think the centering of these hints is hard to read, so lets left-align this)

<img width="540" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/f8dbe5f7-9017-4e2d-a957-5900de269630">

Reply:
I've aligned text to the left, but a bit puzzled how to create story for it without invasive refactoring of the code 🤔 

![image](https://github.com/gravitational/teleport/assets/964862/fea24238-de03-435b-afde-33956ecabef2)

Reply:
Yeah, in the Connect My Computer guide in Discover [I made the timeout configurable through props](https://github.com/gravitational/teleport/blob/8a6e182e8254bdc5f03cbf9cbe01c4b61b90a618/web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx#L53) so that [in the story I can pass just `1`](https://github.com/gravitational/teleport/blob/8a6e182e8254bdc5f03cbf9cbe01c4b61b90a618/web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx#L118-L122). But I don't use `usePingTeleport` there.